src/Microsoft.AspNetCore.SignalR.Redis/RedisHubLifetimeManager.cs

```diff
@@ -52,51 +53,59 @@ public class RedisHubLifetimeManager<THub> : HubLifetimeManager<THub>, IDisposab
             _channels = new RedisChannels(typeof(THub).FullName);
             _protocol = new RedisProtocol(hubProtocolResolver.AllProtocols);
 
-            var writer = new LoggerTextWriter(logger);
             RedisLog.ConnectingToEndpoints(_logger, options.Value.Options.EndPoints, _serverName);
-            _redisServerConnection = _options.Connect(writer);
+        }
 
-            _redisServerConnection.ConnectionRestored += (_, e) =>
+        private async Task EnsureRedisServerConnection()
+        {
+            _redisConnectionSemaphor.Wait();
```

Comment:
Rename to _connectionLock.
WaitAsync for non-blocking.
Double check locking (i.e. if _redisServerConnection is not null then exit method immediately)
Place logic to start connection inside try/finally to ensure _lock.Release is always called